packages/dds/matrix/src/matrix.ts

```diff
@@ -357,16 +357,12 @@ export class SharedMatrix<T extends Serializable = Serializable>
     /** @internal */ public _undoRemoveRows(segment: ISegment) {
         const original = segment as PermutationSegment;
 
-        // (Re)insert the removed number of columns at the original position.
+        // (Re)insert the removed rows at the original position.
         const rowStart = this.rows.getPosition(original);
-        this.insertRows(rowStart, original.cachedLength);
+        const inserted = new PermutationSegment(original.cachedLength, original.start);
+        this.submitRowMessage(this.rows.insertSegmentLocal(rowStart, inserted));
```

Comment:
The "fix" for the issue is to clone the span of row/col handles into the new segment before inserting it.

(Ditto below)